src/app/clusters/service-area-server/service-area-server.cpp

```diff
@@ -0,0 +1,1104 @@
+        ChipLogError(Zcl, "RenameSupportedMap Id %u - map does not exist", aMapId);
+        return false;
+    }
+
+    //  Map name SHALL include readable text that describes the map's name. It cannot be empty string.
+    if (newMapName.empty())
+    {
+        ChipLogError(Zcl, "RenameSupportedMap %u - Name must not be empty string", aMapId);
+        return false;
+    }
+
+    // update the local copy of the map
+    modifiedMap.Set(modifiedMap.mapID, newMapName);
+
+    // Each entry in this list SHALL have a unique value for the Name field.
+    while (mDelegate->GetSupportedMapByIndex(loopIndex, entry))
+    {
+        if (modifiedIndex == loopIndex)
+        {
+            continue; // don't check local modified map against its own list entry
+        }
+
+        if (entry.IsNameEqual(newMapName))
+        {
+            ChipLogError(Zcl, "AddSupportedMap %u - map already exists with same name '%s'", aMapId, entry.GetName().data());
+            return false;
+        }
+
+        ++loopIndex;
+    }
+
+    if (!mDelegate->ModifySupportedMap(modifiedIndex, modifiedMap))
+    {
+        return false;
+    }
+
+    // map successfully renamed
+    NotifySupportedMapsChanged();
+    return true;
+}
+
+bool Instance::ClearSupportedMaps()
+{
+    // does device mode allow this attribute to be updated?
+    if (!mDelegate->IsSupportedMapChangeAllowed())
+    {
+        return false;
+    }
+
+    if (mDelegate->ClearSupportedMaps())
+    {
+        ClearSupportedLocations();
+        NotifySupportedMapsChanged();
+        return true;
+    }
+
+    return false;
+}
+
+//*************************************************************************
+// Selected Locations manipulators
+
+bool Instance::AddSelectedLocation(uint32_t & aSelectedLocation)
+{
+    uint32_t dummyIndex;
+
+    char locationStatusBuffer[kMaxSizeStatusText];
+    MutableCharSpan locationStatusText(locationStatusBuffer);
+
+    // check max# of list entries
+    if (mDelegate->GetNumberOfSelectedLocations() >= kMaxNumSelectedLocations)
+    {
+        ChipLogError(Zcl, "AddSelectedLocation %u - maximum number of entries", aSelectedLocation);
+        return false;
+    }
+
+    // each item in this list SHALL match the LocationID field of an entry on the SupportedLocations attribute's list
+    if (!IsSupportedLocation(aSelectedLocation))
+    {
+        ChipLogError(Zcl, "AddSelectedLocation %u - not a supported location", aSelectedLocation);
+        return false;
+    }
+
+    // each entry in this list SHALL have a unique value
+    if (mDelegate->IsSelectedLocation(aSelectedLocation))
+    {
+        ChipLogError(Zcl, "AddSelectedLocation %u - duplicated location", aSelectedLocation);
+        return false;
+    }
+
+    // Does device mode allow modification of selected locations?
+    if (!mDelegate->IsSetSelectedLocationsAllowed(locationStatusText))
+    {
+        //        ChipLogError(Zcl, "AddSelectedLocation %u - %s", aSelectedLocation, locationStatusText);
```

Comment:
Why is this commented out?  Either remove, or uncomment.

Reply:
Added it back. See [here](https://github.com/project-chip/connectedhomeip/pull/33991/commits/7104c9ead665d1e5fd351a9027bc506aa247d145).